### ISSUE TITLE
Chain id issue when deploying to rinkeby testnet

### ISSUE THREAD

**Issue by paxthemax** [wighawag/buidler-deploy#5]
# Issue 
Deployments fail to rinkeby testnet, with the following error:
```
Error: Network name ("rinkeby") is confusing, chainId is 0x4. Was expecting 0x04
    at DeploymentsManager.getDeploymentsSubPath (/home/pax/src/buidler-deploy-ts-test/node_modules/buidler-deploy/src/DeploymentsManager.ts:662:15)
    at DeploymentsManager.loadDeployments (/home/pax/src/buidler-deploy-ts-test/node_modules/buidler-deploy/src/DeploymentsManager.ts:279:12)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

## Steps to reproduce:
I've taken this [test repo](https://github.com/wighawag/buidler-deploy-ts-test), and installed the latest verion of buidler-deploy (`yarn install buidler-deploy@latest`).

Here is the `package.json` after the installation:
```
{
  "devDependencies": {
    "@nomiclabs/buidler": "^1.2.0",
    "@types/chai": "^4.2.11",
    "@types/mocha": "^7.0.2",
    "@types/node": "^13.11.0",
    "buidler-deploy": "^0.3.3",
    "buidler-ethers-v5": "^0.2.0",
    "chai": "^4.2.0",
    "cross-env": "^7.0.2",
    "dotenv": "^8.2.0",
    "ethers": "^5.0.0-beta.180",
    "ts-node": "^8.8.2",
    "typescript": "^3.8.3"
  },
  "scripts": {
    "test": "buidler test",
    "run:rinkeby": "cross-env BUIDLER_NETWORK=rinkeby ts-node --files",
    "deploy:rinkeby": "buidler --network rinkeby deploy",
    "dev": "buidler listen --export contractsInfo.json"
  }
}
```
Running `yarn deploy:rinkeby` results in the above error.

## Notes
The bug does not appear on `buidler-deploy` version v0.2.1, only on the upgrade.

It would appear that the issue comes from a mismatch between the types of the chainID in the Buidler env (number) and the DeployManager (hex string).

Also, the error message is confusing (oh the irony :D)

**Comment by wighawag** [comment]
Thanks for reporting. I am thorn in using hex value for chainIds. This is what the raw `eth_chainId` return but most tool use number to represent chainId 
And the issue I guess is that some node do not return even number length hex string.

Probably going to update to use number or decimal string to follow the current tooling convention.

**Comment by wighawag** [comment]
#I pushed a fix on 0.3.4 that should solve the issue

**Comment by paxthemax** [comment]
Confirmed, deployment to rinkeby works. Thanks! :+1:
